File: app/src/main/java/org/digitalcampus/oppia/model/CoursesRepository.java
package org.digitalcampus.oppia.model;

import android.content.Context;

import org.digitalcampus.oppia.database.DbHelper;
import org.digitalcampus.oppia.application.SessionManager;

import java.util.List;

public class CoursesRepository {

    public List<Course> getCourses(Context ctx){
        DbHelper db = DbHelper.getInstance(ctx);
        long userId = db.getUserId(SessionManager.getUsername(ctx));
        return db.getCoursesForUser(userId);
    }

    public Activity getActivityByDigest(Context ctx, String digest){
        DbHelper db = DbHelper.getInstance(ctx);
        return db.getActivityByDigest(digest);
    }

    public Course getCourse(Context ctx, long courseID, long userID){
        DbHelper db = DbHelper.getInstance(ctx);
        return db.getCourse(courseID, userID);
    }

    public Course getCourseByShortname(Context ctx, String shortname, long userID){
        DbHelper db = DbHelper.getInstance(ctx);
        long courseId = db.getCourseIdByShortname(shortname);
        if (courseId != -1) {
            return db.getCourse(courseId, userID);
        }
        return null;
    }
}

File: app/src/main/java/org/digitalcampus/oppia/model/CoursesRepository.kt
package org.digitalcampus.oppia.model

import android.content.Context
import org.digitalcampus.oppia.application.SessionManager
import org.digitalcampus.oppia.database.DbHelper

class CoursesRepository {
    fun getCourses(ctx: Context?): List<Course> {
        val db = DbHelper.getInstance(ctx)
        val userId = db.getUserId(SessionManager.getUsername(ctx))
        return db.getCoursesForUser(userId)
    }

    fun getActivityByDigest(ctx: Context?, digest: String?): Activity {
        val db = DbHelper.getInstance(ctx)
        return db.getActivityByDigest(digest)
    }

    fun getCourse(ctx: Context?, courseID: Long, userID: Long): Course {
        val db = DbHelper.getInstance(ctx)
        return db.getCourseWithProgress(courseID, userID)
    }

    fun getCourseByShortname(ctx: Context?, shortname: String?, userID: Long): Course? {
        val db = DbHelper.getInstance(ctx)
        val courseId = db.getCourseIdByShortname(shortname, userID)
        return if (courseId != -1L) {
            db.getCourseWithProgress(courseId, userID)
        } else null
    }
}